src/huggingface_hub/hf_api.py

```diff
@@ -9673,6 +9737,82 @@ def list_user_following(self, username: str, token: Union[bool, str, None] = Non
         ):
             yield User(**followed_user)
 
+    def list_papers(
+        self,
+        *,
+        date: Optional[str] = None,
+        query: Optional[str] = None,
+    ) -> Iterable[PaperInfo]:
+        """
+        List daily papers on the Hugging Face Hub, given a date or a search query.
+
+        Args:
+            date (`str`, *optional*):
+                The date to retrieve papers for, in the format 'YYYY-MM-DD'.
+                If provided, returns papers submitted on this date.
+            query (`str`, *optional*):
+                A search query string to find papers.
+                If provided, returns papers that match the query.
+        Returns:
+            `Iterable[PaperInfo]`: A list of [`PaperInfo`] objects.
+
+        Raises:
+            [`HTTPError`](https://requests.readthedocs.io/en/latest/api/#requests.HTTPError):
+                HTTP 400 if the date is invalid.
+            [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError):
+                If neither `date` nor `query` is provided.
+
+        Example usage with the `date` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers submitted on a specific date
+        >>> api.list_papers(date="2024-09-17")
+        ```
+
+        Example usage with the `query` argument:
+
+        ```python
+        >>> from huggingface_hub import HfApi
+
+        >>> api = HfApi()
+
+        # List all papers with "attention" in their title
+        >>> api.list_papers(query="attention")
+        ```
+        """
+        if date is None and query is None:
+            raise ValueError("Provide one of `date` or `query`.")
+        if date:
+            r = get_session().get(f"{constants.ENDPOINT}/api/daily_papers?date={date}")
+        elif query:
+            r = get_session().get(f"{constants.ENDPOINT}/api/papers/search?q={query}")
```

Comment:
```suggestion
        path = f"{self.endpoint}/api/daily_papers"
        params = {}
        if date:
            params["date"] = date
        if query:
            params["q"] = query
        r = get_session().get(path, params=params)
```
Sorry to get back on this 🙈 I think we can use the endpoint `/api/daily_papers` for both `date` and `query`. Actually, I tried passing both filters, it does not seem to return the daily papers of the specified date that match the query. It seems the query takes precedence on the date